### PR TITLE
[ActionSheet] Allow clients to align titles if only some actions have images.

### LINKED ISSUE #7484: [ActionSheet] Internal issue: b/130018562
This was filed as an internal issue. If you are a Googler, please visit [b/130018562](http://b/130018562) for more details.

---
#### Internal data
- Associated internal bug: [b/130018562](http://b/130018562)

### PR DESCRIPTION
**Pull request by codeman7**
This adds functionality to MDCActionSheetController to allow clients to align the titles even if some actions have images and others don't. If a client sets this flag to `YES` but none of the actions have images then we need to make sure that we don't add padding to the cells, that check is in the setter. Additionally the setter reloads the data so that if a client sets this property after and _ActionSheet_ has been presented then the table would update. Test were added for the `init` for both `MDCActionSheetController` and `MDCActionSheetItemTableViewCell` but only for the property added here, this is to minimize the changes in this PR. I have opened up # to track adding test for all default properties.

Closes #7484

### CODE REVIEW

File: components/ActionSheet/src/MDCActionSheetController.h [hunk 1]
```diff
@@ -181,6 +181,15 @@ __attribute__((objc_subclassing_restricted)) @interface MDCActionSheetController
...
 @property(nonatomic) UIImageRenderingMode imageRenderingMode;
 
+/**
+ Determines if all titles are aligned or if the titles are aligned based on if there is an image
+ assocated with the given @c MDCActionSheetAction. Setting this to @c YES will align all titles
+ regardless of if all @c actions have images or not.
+
+ @note Defaults to @c NO.
+ */
+@property(nonatomic, assign) BOOL alignTitles;
```
Comment (jverkoey) [inline components/ActionSheet/src/MDCActionSheetController.h]: Consider `alwaysAlignTitleLeadingEdges` to make it more clear what aspect of the titles are being aligned, and that there's a degree of unconditionality at play.

File: components/ActionSheet/src/MDCActionSheetController.h [hunk 2]
```diff
@@ -181,6 +181,15 @@ __attribute__((objc_subclassing_restricted)) @interface MDCActionSheetController
  */
 @property(nonatomic) UIImageRenderingMode imageRenderingMode;
 
+/**
+ Determines if all titles are aligned or if the titles are aligned based on if there is an image
```
Comment (jverkoey) [inline components/ActionSheet/src/MDCActionSheetController.h]: Consider:
```
Determines the alignment behavior of all title leading edges.

When YES, all title leading edges will be aligned with one another.

When NO, each title will be positioned individually based on whether it has an image or not.

Defaults to NO.
```

File: components/ActionSheet/src/MDCActionSheetController.m [hunk 3]
```diff
@@ -416,6 +424,20 @@ - (void)setImageRenderingMode:(UIImageRenderingMode)imageRenderingMode {
...
+- (void)setAlignTitles:(BOOL)alignTitles {
+  _alignTitles = alignTitles;
+  BOOL addPaddingToCells = NO;
+  // Check to make sure atleast one action has an image. If not then all actions will align already
+  // and we don't need to add padding.
+  for (MDCActionSheetAction *action in self.actions) {
+    if (action.image) {
+      addPaddingToCells = YES;
+    }
+  }
```
Comment (jverkoey) [inline components/ActionSheet/src/MDCActionSheetController.m:440]: Please extract this for loop to a separate private method and invoke it. E.g.:

```objc
- (BOOL)anyActionHasAnImage {
  for (MDCActionSheetAction *action in self.actions) {
    if (action.image) {
      return YES;
    }
  }
  return NO;
}
self.addLeadingPaddingToCell = [self anyActionHasAnImage];
```

File: components/ActionSheet/src/MDCActionSheetController.h (hunk 2 above)
Comment (codeman7) [reply components/ActionSheet/src/MDCActionSheetController.h]: Done

File: components/ActionSheet/src/MDCActionSheetController.m (hunk 3 above)
Comment (codeman7) [reply components/ActionSheet/src/MDCActionSheetController.m:440]: Done

File: components/ActionSheet/src/MDCActionSheetController.h (hunk 1 above)
Comment (codeman7) [reply components/ActionSheet/src/MDCActionSheetController.h]: Done

File: components/ActionSheet/src/MDCActionSheetController.m [hunk 4]
```diff
@@ -416,6 +424,23 @@ - (void)setImageRenderingMode:(UIImageRenderingMode)imageRenderingMode {
...
 
+- (void)setAlwaysAlignTitleLeadingEdges:(BOOL)alignTitles {
+  _alwaysAlignTitleLeadingEdges = alignTitles;
+  // Check to make sure atleast one action has an image. If not then all actions will align already
+  // and we don't need to add padding.
+  self.addLeadingPaddingToCell = [self anyActionHaveImage];
+  [self.tableView reloadData];
+}
+
+- (BOOL)anyActionHaveImage {
```
Comment (jverkoey) [inline components/ActionSheet/src/MDCActionSheetController.m]: `anyActionHasAnImage` or `anyActionHasImage`

File: components/ActionSheet/src/MDCActionSheetController.m [hunk 5]
```diff
@@ -416,6 +424,23 @@ - (void)setImageRenderingMode:(UIImageRenderingMode)imageRenderingMode {
   [self.tableView reloadData];
 }
 
+- (void)setAlwaysAlignTitleLeadingEdges:(BOOL)alignTitles {
+  _alwaysAlignTitleLeadingEdges = alignTitles;
+  // Check to make sure atleast one action has an image. If not then all actions will align already
```
Comment (jverkoey) [inline components/ActionSheet/src/MDCActionSheetController.m]: "at least"

**Review by jverkoey — APPROVED** [review]

File: components/ActionSheet/src/MDCActionSheetController.m (hunk 4 above)
Comment (codeman7) [reply components/ActionSheet/src/MDCActionSheetController.m]: Done

File: components/ActionSheet/src/MDCActionSheetController.m (hunk 5 above)
Comment (codeman7) [reply components/ActionSheet/src/MDCActionSheetController.m]: Done

**Pull request merged**